src/pkg/extensions/extensions.go

```diff
@@ -0,0 +1,126 @@
+package extensions
+
+import (
+	"context"
+	"io"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+// Extension client interface
+type CorsoItemExtension interface {
+	io.ReadCloser
+}
+
+type CorsoItemExtensionFactory func(
+	context.Context,
+	io.ReadCloser,
+	details.ItemInfo,
+	*details.ExtensionInfo,
```

Comment:
imo, this should be an interface, not a pointer to the struct itself.  That way you can provide the user something simple like this, and no one needs to worry about concrete implementation.  Also useful for testing, in this case.
```
type ExtensionKVStorer interface {
func StoreExtensionKV(k string, v any)
}
```

Reply:
Great suggestion! I did have later plans on exposing `details.ItemInfo` via a getter and `ExtensionInfo` via a setter interface(one you recommended above). But I wanted to get out something quick.

Let me add the `ExtensionKVStorer` right away, as it's a great pattern &  will save extension authors some refactoring time later.